src/trie.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+from collections import deque, OrderedDict
+
+
+class Trie(object):
+    def __init__(self, iterable=None):
+        self.root = OrderedDict()
+        if iterable:
+            for i in iterable:
+                self.insert(i)
+
+    def insert(self, token):
+        start = self.root
+        for letter in token:
+            start = start.setdefault(letter, OrderedDict())
+        start['#'] = '#'
```

Comment:
Since you're using `'#'` as your end of word token throughout your code, it would be a great idea and correct python usage to set a global constant e.g. `TERM_CHAR = '#'`